sklearnex/_device_offload.py

```diff
@@ -73,26 +74,42 @@ def dispatch(obj, method_name, branches, *args, **kwargs):
         patching_status.write_log(queue=q)
         return branches[backend](obj, *hostargs, **hostkwargs, queue=q)
     if backend == "sklearn":
-        patching_status.write_log()
-        return branches[backend](obj, *hostargs, **hostkwargs)
+        if "array_api_dispatch" in get_config() and get_config()["array_api_dispatch"]:
+            # TODO:
```

Comment:
Is this TODO supposed to be solved in this PR?  I think a comment or two in the code might be nice for the progression of the data from an array_api type to the onedal_*_supported to the fallback or to onedal branch may be useful. Its hard for me to walk through the logic.  It seems in all cases data is moved from  array_api supported types, but in fallback it is only sent as the original if the config is set. If thats true it may be worth making a note since there isn't much locality to the logic.

Reply:
Note is done. Also a bit update conditions for using raw inputs.